### DISCUSSION TITLE
Package weight parameter missing (crucial to calculate shipping cost!)

### DISCUSSION THREAD

**Discussion by TVBZ** [PrestaShop/PrestaShop#35063] [Ideas]
#### Is your feature request related to a problem?

The request is pretty basic and simple. We need to be able to add a package weight somewhere in configuration.
Currently, if I would sell and send some products at a total of 9.8 KG, but the box I use to send it is 0.3 KG, the total weight would be 10.1 KG. My shipping cost for packages < 10 KG and > 10 KG is not the same. Because PS doesn't add my box weight to total, it calculates the wrong shipping price.

#### Describe the solution you'd like

Add a parameter somewhere so an additional weight can be added to the total weight before calculation of shipping cost.
There are handling charges in **shipping -> preferences**,  there ~~could~~ should also be **handling weight**.

#### Alternatives you've considered

I found an addon that has this purpose (among other things) but it seams illogical to pay €50 euro for such basic function, risking to have to pay the license again and again for each PS update in the future.

#### Additional context

There is lots of chatter about this topic on PS forums, dating back to PS 1.3.
It's hard to understand why it's not yet supported. 
As a PS newbie, this raises some red flags to continue using PS.

Thanks for your consideration

[a comment by an automated account is not shown]

**Comment by sarahdib** [comment]
@PrestaShop/prestashop-product-team What do you think about this improvement ? add the package weight to calculate shipping cost ?

**Comment by Hlavtox** [comment]
Why don't you just add it to every product? 0.1kg to every product, for example.

**Comment by TVBZ** [comment]
@Hlavtox That seams a bit overkill. What if your shop has 2000 products and you want to change package weight?

For now, I worked around this issue by simply substracting package weight from carriers.
Where a price was set to up to 10,000 grams, it's now set to up to 9,899 grams.
I think this is more workable than adding it to all products. In most cases, theres will be a lot less carriers than products.

Still, I think it would be good addition to add a core package weight parameter.

**Comment by Hlavtox** [comment]
@TVBZ Another solution! :) I can do this for you in 5 minutes, export it from DB, change in Excel, import back.

How would you do the package weights? You would have to have dozens of them, depending on the weight and size of the product. Weight of the box will not be the same every time.

In real life, sometimes you pay some of the shipping, sometimes you make something on the shipping, sometimes you give shippinf for free. Don't make too much of a relationship between the customer's shipping price and the price the shipping company invoices you. That's just merchants life, IMHO. 😎

**Comment by JBWModules** [comment]
A simple package weight is not enough as it might change depending on the content - bigger box, more weight. But how to know which products fit best into which package? Additonally you might have to wrap some inside the box with additional package materials etc. You might even have to split deliveries into severals packages and delivieries, thats something Presta does not support out of the box but some modules could do.

**Comment by TVBZ** [comment]
@Hlavtox True. But why do it in Excel if you're allready in phpMyAdmin or similar? :)
I understand. In my case, the package varies depending on order size, but it's never more than 100 grams.

@JBWModules I agree. But it's not allways the case. If a certain product needs special packaging and it makes a big difference, I would add it to the product itself. Otherwise, I don't see the box becoming much heavier.

I guess calculating the package weight into the carrier settings is a good enought way to do it. Thanks for the insights. :)

**Comment by DeFault-1** [comment]
There should be a new column in the product weight ranges to enter manually the total packaging weight (wrapping plus box weight) for every weight range. I always estimate the wrapping materials weight as 10% of the product weight. As far as the box weight is concerned, in my case the carrier is selling a range of standard postal boxes, we can decide which particular box will be used for a certain product weight range. This is not so difficult indeed.

**Comment by RogerAndersson** [comment]
I have the same need to add weight for product packaging.
Maybe possible to add weight depending on max product size in cart.

**Comment by Hlavtox** [comment]
@RogerAndersson @DeFault-1 @TVBZ

The section in product options is called "Delivery" and the input group has header "Package dimensions". You should enter the product **box** dimensions and weight. If you have 2000 products, well, you should have done it right from the start OR, don't tell me that global change would take more than 10 minutes. 😄

And if you from whatever reason don't want to change your product box weights, you can add the package weight fairly simple by overriding just one function in Carrier.php. 👍 Or, prestashop has proper hooks to easily make your own carrier module, and it will use your shipping calculations instead of the default ranges. You can also add extra weight there.

Or there is even a hook in the function: `$price_by_weight = Hook::exec('actionDeliveryPriceByWeight', array('id_carrier' => $id_carrier, 'total_weight' => $total_weight, 'id_zone' => $id_zone));`. You can just make a simple module and hook into there and just modify the price as you wish.

![Výstřižek](https://user-images.githubusercontent.com/6097524/107860762-6f0d3c00-6e41-11eb-95b6-83832c659d78.JPG)

**Comment by alfonsolr09** [comment]
I think it is silly since that product weight is not shown in the F.O.
 The weight of the product + package should be written in the "weight" box as it is a transport control.

also how much does a package weigh? I don't think weight is that important ... it is better to even raise 1kg the weight to all products

**Comment by arirangz** [comment]
Hi,

Is there any plan to add this field?

Thanks

**Comment by cdigruttola** [comment]
Hi @TVBZ,
I created this free module for the imprevement https://github.com/cdigruttola/packageweight

**Comment by HartLarsson** [comment]
> Hi @TVBZ, I created this free module for the imprevement https://github.com/cdigruttola/packageweight

can you explain what is doing you rmodule?

**Reply by cdigruttola** [reply]
It helps you to add the package weight for every RangeWeight you set. You can test it on this link https://cdigruttola.it/prestashop/admin712ixfyo12kcvreyjxf. To access it please use the credential demo@demo.com demodemo

**Comment by HartLarsson** [comment]
the package weight is interestign but what about volumetric calculation based on 3 parameters we already have inside the productr L x  H x W.
Missing only a cm3/Kg or m3/Kg parameter to specify in the currier data

**Reply by cdigruttola** [reply]
Hi, if you are referring to the module I know but it cuold be a starting point to add these calculation using volumetric parameters.

**Comment by MatShir** [comment]
Reopening the issue without an explanation for its initial closure. I also transformed it into a discussion. This is where we will now submit the feature request.
